### ISSUE TITLE
 Why not add mac adress address support?

### ISSUE THREAD

**Issue by Likeeit** [HerculesWS/Hercules#1734]
We all know of such need in the emulator and how important that would be. Here's the question: Why not add mac adress address support?
NOTE: This was implemented here in Brazil, in an emulator that is fork of Hercules.
https://github.com/brAthena/brAthena/commit/d1ebe6dd5424bd92ec6bbd0e1b9ff033105f80f7

**Comment by mekolat** [comment]
mac addresses can very easily be changed, and nothing prevents a player from intercepting the packet in transit to change the mac address that is sent to the server

**Comment by Smokexyz** [comment]
If it is part of the login packet from the client, it should be handled by the server. But as @mekolat says, it is impossible to truly fingerprint a user.

**Comment by Likeeit** [comment]
Sirs, I believe you have misunderstood. What was said in the text above is true, but this should not be taken only to the MAC, but rather to the whole system. Every system is mocking, however good it may be, people may find it not, but they just do not know how to do it. The intention of adding the MAC, is to have one more option to 'defend' the abusers.

**Comment by mekolat** [comment]
IMO the mac address does not provide any additional protection, and opens the door to bad scripting practices that relies on user-submitted data (the packet can be forged) as a "security" measure

**Comment by Asheraf** [comment]
Also gravity client does not support mac address so this should be kept as
a custom mod

**Comment by Likeeit** [comment]
One option would be to add the MAC as an additional means of protection, where it would not have to be active by default, would be at the discretion of each player to use or not if he wanted to.

**Comment by mekolat** [comment]
It offers absolutely ZERO protection

**Comment by 4144** [comment]
@mekolat in ro world most protections works like this. they "protecting" and "obfuscating" things in client.
From other side this kind of protections can stop noobs.

About mac address. In hercules exists SSO like packet with mac address, but mac field here unused. And i not sure is any client can send this packet or not.

**Comment by MishimaHaruna** [comment]
@4144 The client can send the 0x0825 login packet (`PACKET_ID_CA_SSO_LOGIN_REQ`). I believe it's the main packet that kRO uses to log in, and the client sends it when using a korean langtype and isn't modified to re-introduce the legacy login screen.
I believe the MAC field is set to zero in many cases though (or at least, it is when the client is run through wine - I don't have a windows machine equipped with a Ragnarok client to see how it behaves there). This makes the MAC field even more useless than it normally would.

**Comment by dastgir** [comment]
status shouldn't be wontfix maybe?

kRO records the MacAddress if we use kRO client to login and we don't even save the mac for 0x0825 packet

**Comment by MishimaHaruna** [comment]
I don't believe the MAC address is of any use, but okay, since the field exists there, we can accept pull requests that handle it.

The reasons why I won't waste time on it unless a pull request comes are (summarized from the above):

- The MAC address is very easy to spoof, and doesn't really identify an user
- From my experience, it's always set to zero
